tests/drivers/crc/src/main.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2024 Brill Power Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/crc.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+/* Test cases for CRC_STM32 driver */
+
+#define WAIT_THREAD_STACK_SIZE 1024
+#define WAIT_THREAD_PRIO       -10
+
+static void wait_thread_entry(void *a, void *b, void *c);
+
+K_THREAD_STACK_DEFINE(wait_thread_stack_area, WAIT_THREAD_STACK_SIZE);
+struct k_thread wait_thread_data;
+
+/**
+ * 1) Take the semaphore
+ * 2) Sleep for 50 ms (to allow ztest main thread to attempt to acquire semaphore)
+ * 3) Give the semaphore
+ */
+static void wait_thread_entry(void *a, void *b, void *c)
+{
+	UNUSED(a);
+	UNUSED(b);
+	UNUSED(c);
```

Comment:
s/UNUSED|ARG_UNUSED

IIRC UNUSED is provided by STM32Cube HAL and should not be used in Zephyr